py/parse.c

```diff
@@ -427,6 +417,25 @@ STATIC mp_obj_t mp_parse_node_convert_to_obj(mp_parse_node_t pn) {
 }
 #endif
 
+STATIC bool parse_node_is_const_bool(mp_parse_node_t pn, bool value) {
+    // Returns true if 'pn' is a constant whose boolean value is equivalent to 'value'
+    #if MICROPY_COMP_CONST_TUPLE || MICROPY_COMP_CONST
+    return mp_parse_node_is_const(pn) && !!mp_obj_is_true(mp_parse_node_convert_to_obj(pn)) == value;
```

Comment:
Is the `!!` needed here?  `mp_obj_is_true` returns a bool (0 or 1) so it should be OK without it (and may reduce code size without the `!!`).

Reply:
You're right, not needed. Fixed, thanks for noticing!

(No difference in code size unfortunately, I was curious so I checked - as `mp_obj_is_true` has type return bool the compiler already knows the values are constrained to 0/1.)